src/writer/async_tokio.rs

```diff
@@ -340,3 +484,99 @@ mod indentation_async {
         );
     }
 }
+
+#[cfg(test)]
+mod element_writer {
+    use super::*;
+    use crate::events::*;
+    use pretty_assertions::assert_eq;
+
+    #[tokio::test]
+    async fn empty() {
```

Comment:
Hm... maybe move all those tests to doc comments of corresponding methods? I think, this will be useful. This one for `write_empty_async`